File: app/backend/java/payment-service/src/main/java/com/srbms/paymentservice/strategy/CardPaymentProcessor.java
package com.srbms.paymentservice.strategy;

import com.srbms.paymentservice.dto.PaymentRequestDTO;
import com.srbms.paymentservice.entity.Payment;
import com.srbms.paymentservice.entity.PaymentMode;
import com.srbms.paymentservice.entity.PaymentStatus;
import org.springframework.stereotype.Component;

import java.math.BigDecimal;
import java.util.UUID;

@Component
public class CardPaymentProcessor implements PaymentProcessorStrategy {

    @Override
    public boolean supports(PaymentMode paymentMode) {
        return paymentMode == PaymentMode.CARD;
    }

    @Override
    public Payment process(Payment payment, PaymentRequestDTO requestDTO) {
        // Simulating credit/debit card gateway payment validation & charge
        payment.setPaymentMode(PaymentMode.CARD);
        payment.setGatewayRef("CARD_GW_" + UUID.randomUUID().toString().substring(0, 8).toUpperCase());
        payment.setPaymentStatus(PaymentStatus.SUCCESS);
        return payment;
    }

    @Override
    public Payment refund(Payment payment, BigDecimal amount, String reason) {
        payment.setPaymentStatus(PaymentStatus.REFUNDED);
        payment.setGatewayRef("CARD_RFD_" + UUID.randomUUID().toString().substring(0, 8).toUpperCase());
        return payment;
    }
}

File: app/backend/.net/SmartRide.PaymentService/Strategies/CardPaymentProcessor.cs
using SmartRide.PaymentService.Models.DTOs;
using SmartRide.PaymentService.Models.Entities;
using SmartRide.PaymentService.Models.Enums;

namespace SmartRide.PaymentService.Strategies;

public class CardPaymentProcessor : IPaymentProcessorStrategy
{
    public bool Supports(PaymentMode paymentMode)
    {
        return paymentMode == PaymentMode.CARD;
    }

    public Task<Payment> ProcessAsync(Payment payment, PaymentRequestDTO requestDTO)
    {
        payment.PaymentMode = PaymentMode.CARD;
        payment.GatewayRef = "CARD_GW_" + Guid.NewGuid().ToString("N")[..8].ToUpper();
        payment.PaymentStatus = PaymentStatus.SUCCESS;
        return Task.FromResult(payment);
    }

    public Task<Payment> RefundAsync(Payment payment, decimal? amount, string reason)
    {
        payment.PaymentStatus = PaymentStatus.REFUNDED;
        payment.GatewayRef = "CARD_RFD_" + Guid.NewGuid().ToString("N")[..8].ToUpper();
        return Task.FromResult(payment);
    }
}
